### DISCUSSION TITLE
list display vs MutableSequence argument

### DISCUSSION THREAD

**Discussion by pinterior** [microsoft/pyright#7497] [Q&A] [answered]
situation: passing to list[T] to function takes list[T | U] or MutableSequence[T | U].

consider two functions:
```python
def foo(x: list[int | str]):
    ...

def bar(x: MutableSequence[int | str]):
    ...
```

when passing list of int as list display, only `bar([1])` causes error.
is it intened?

```python
foo([1])  # ok
bar([1])  # error: Argument of type "list[int]" cannot be assigned to parameter "x" of type "MutableSequence[int | str]" in function "foo" (reportArgumentType)
```

if manually typed, it goes fine
```python
b: list[int | str] = [1]
foo(b)  # ok
bar(b)  # ok
```

**Comment by erictraut** [comment, marked as answer]
Thanks for bringing this to my attention.

Through bidirectional type inference, neither of these should result in an error. I've created a [bug report](https://github.com/microsoft/pyright/issues/7498) to track this.
